src/sentry/static/sentry/app/views/dashboards/exploreWidget.jsx

```diff
@@ -38,57 +39,153 @@ class ExploreWidget extends React.Component {
     });
   };
 
+  // TODO(discover1): Can be removed when Discover1 is deprecated
+  // Copied from https://github.com/getsentry/sentry/blob/8d31f8651558b3f9a5f65dc45e0f439c5ac19d55/src/sentry/static/sentry/app/components/sidebar/index.jsx#L230-L278
+  getDiscoverFlags(organization) {
+    const flags = {
+      discover1: false,
+      discover2: false,
+      events: false,
+    };
+
+    // Bail as we can't do any more checks.
+    if (!organization || !organization.features) {
+      return flags;
+    }
+
+    // Localstorage returns either null, 1 or 2. Default to 2.
+    const version = String(localStorage.getItem('discover:version') || 2);
+    const features = organization.features;
+
+    if (features.includes('discover-basic')) {
+      // If there is no opt-out state show discover2
+      if (!version || version === '2') {
+        flags.discover2 = true;
+      }
+      // User wants discover1
+      if (version === '1') {
+        flags.discover1 = true;
+        flags.events = true;
+      }
+      return flags;
+    }
+
+    // If an account has the old features they continue to have
+    // access to them.
+    if (features.includes('discover')) {
+      flags.discover1 = true;
+    }
+    if (features.includes('events')) {
+      flags.events = true;
+    }
+
+    // If an organization doesn't have events, or discover-basic
+    // Enable the tab so we can show an upsell state in saas.
+    if (!flags.events) {
+      flags.discover2 = true;
+    }
+
+    return flags;
+  }
+
+  renderActionToDiscover1(query, flags) {
+    // Hide if preference is Discover2
+    if (flags.discover2) {
+      return null;
+    }
+
+    if (!flags.discover1) {
+      return null;
+    }
+
+    return (
+      <ExploreAction
+        to={this.getExportToDiscover(query)}
+        title={
+          flags.discover1
+            ? t('Explore data in Discover')
+            : t('You do not have access to Discover')
+        }
+      >
+        <InlineSvg src="icon-discover" />
+      </ExploreAction>
+    );
+  }
+
+  renderActionToDiscover2(query, flags) {
+    // If Discover1 is the preference, do not show
+    if (flags.discover1 && flags.events && !flags.discover2) {
+      return null;
+    }
+
+    return (
+      <ExploreAction
+        to={flags.discover2 ? this.getExportToDiscover(query, true) : ''}
+        href={!flags.discover2 ? 'https://docs.sentry.io/workflow/discover2/' : ''}
```

Comment:
Does this work with both `to` and `href` defined? (I'm not sure what `ExploreAction` is, assuming it's a Button)

Reply:
Yes, it will work with both defined.

https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/components/button.tsx#L278-L284